lib/codegen/src/simple_preopt.rs

```diff
@@ -536,11 +542,279 @@ fn simplify(pos: &mut FuncCursor, inst: Inst) {
     }
 }
 
+struct BranchOptInfo {
+    br_inst: Inst,
+    cmp_arg: Value,
+    destination: Ebb,
+    args: ValueList,
+    kind: BranchOptKind,
+}
+
+enum BranchOptKind {
+    EqualZero,
+    NotEqualZero,
+}
+
+fn branch_opt(pos: &mut FuncCursor, inst: Inst) {
+    let info = match pos.func.dfg[inst] {
+        InstructionData::BranchInt {
+            opcode: Opcode::Brif,
+            cond: br_cond,
+            destination,
+            ref args,
+        } => {
+            let first_arg = {
+                let args = pos.func.dfg.inst_args(inst);
+                args[0]
+            };
+            if let ValueDef::Result(iconst_inst, _) = pos.func.dfg.value_def(first_arg) {
```

Comment:
Please do this like the following to avoid that immense amount of indentation:

```rust
let iconst_inst = if let ValueDef::Result(iconst_inst, _) = pos.func.dfg.value_def(first_arg) {
    iconst_inst
} else {
    return;
};
```